devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -537,6 +542,10 @@ private void triggerPrepare(boolean test) throws MojoExecutionException {
                 Collections.singletonMap("mode", LaunchMode.DEVELOPMENT.name()));
     }
 
+    private void initClassIndexes() throws MojoExecutionException {
+        executeIfConfigured(ORG_JBOSS_JANDEX, JANDEX_MAVEN_PLUGIN, "jandex", Collections.emptyMap());
```

Comment:
So this would _always_ execute the jandex-plugin in case it's configured, _even_ if `process-classes` or an even higher goal was added by the user before `quarkus:dev` (see also https://github.com/quarkusio/quarkus/pull/22868#discussion_r786980254).
AFAICS, the only way to solve this properly with that current "assumption-driven" approach, is to add:
- `POST_PROCESS_CLASSES_PHASES`
- `POST_PROCESS_TEST_CLASSES_PHASES`

analogous to the existing `POST_COMPILE_PHASES`/`POST_TEST_COMPILE_PHASES`.
(hint: `POST_PROCESS_CLASSES_PHASES` is a subset of `POST_COMPILE_PHASES`, so you don't need to duplicate everything)

TBH, I would simply add `processClassesNeeded` or similar to `triggerCompile()` (and calculate the flag in `handleAutoCompile()`).

Reply:
@aloubyansky WDYT about ⬆️? I do think we shouldn't execute jandex-plugin all the time.

Reply:
I'm not a maven user, but is it common to list all goals or does user mostly run ˋquarkus:dev` directly ?

Reply:
The whole idea is that they should run `mvn quarkus:dev` and that's it.

Reply:
I agree @famod 

We are not that close to simply `mvn quarkus:dev` yet, assuming we are taking into account multi-module projects.

Reply:
Ok, I will apply this change to the code